inference-engine/src/mkldnn_plugin/utils/verbose.cpp

```diff
@@ -0,0 +1,145 @@
+        case GREEN:  colorCode = green;
+            break;
+        case YELLOW: colorCode = yellow;
+            break;
+        case BLUE:   colorCode = blue;
+            break;
+        case PURPLE: colorCode = purple;
+            break;
+        case CYAN:   colorCode = cyan;
+            break;
+        default:     colorCode = reset;
+            break;
+        }
+
+        return colorCode + str + reset;
+    };
+
+    auto shift = [&](int size) {
+        if (written < 0 || written_total + size > CPU_VERBOSE_DAT_LEN) {
+            const char* errorMsg = "# NOT ENOUGHT BUFFER SIZE #";
+            snprintf(portsInfo, strlen(errorMsg) + 1, "%s", errorMsg);
+            written_total = strlen(errorMsg);
+            return;
+        }
+
+        written_total += size;
+    };
+
+    auto formatMemDesc = [&](const dnnl_memory_desc_t& desc, std::string& prefix) {
+        prefix = colorize(BLUE, prefix);
+        written = snprintf(portsInfo + written_total, CPU_VERBOSE_DAT_LEN - written_total, " ");
+        shift(written);
+        written = snprintf(portsInfo + written_total, CPU_VERBOSE_DAT_LEN - written_total, "%s", prefix.c_str());
+        shift(written);
+        written = dnnl_md2fmt_str(portsInfo + written_total, CPU_VERBOSE_DAT_LEN - written_total, &desc);
+        shift(written);
+        written = snprintf(portsInfo + written_total, CPU_VERBOSE_DAT_LEN - written_total, ":");
+        shift(written);
+        written = dnnl_md2dim_str(portsInfo + written_total, CPU_VERBOSE_DAT_LEN - written_total, &desc);
+        shift(written);
+    };
+
+    for (int i = 0; i < node->getParentEdges().size(); i++) {
+        std::string prefix("src:" + std::to_string(i) + ':');
+        formatMemDesc(node->getParentEdgeAt(i)->getMemory().GetDescriptor().data, prefix);
+    }
+
+    for (int i = 0; i < node->getChildEdges().size(); i++) {
+        std::string prefix("dst:" + std::to_string(i) + ':');
+        formatMemDesc(node->getChildEdgeAt(i)->getMemory().GetDescriptor().data, prefix);
+    }
+
+    std::string post_ops;
+    if (!node->getFusedWith().empty()) {
+        post_ops += "post_ops:'";
+        for (const auto& fusedNode : node->getFusedWith()) {
+            post_ops.append(colorize(GREEN, fusedNode->getName())).append(":")
+                .append(colorize(CYAN, NameFromType(fusedNode->getType()))).append(":")
+                .append(algToString(fusedNode->getAlgorithm()))
+                .append(";");
+        }
+        post_ops += "'";
+    }
+
+    const std::string nodeName = colorize(GREEN, node->getName());
+    const std::string nodeType = colorize(CYAN, NameFromType(node->getType()));
+    const std::string nodeAlg  = algToString(node->getAlgorithm());
+    const std::string nodePrimImplType =  impl_type_to_string(node->getSelectedPrimitiveDescriptor()->getImplementationType());
+    const auto duration = node->PerfCounter().duration().count();
+
+    // Example:
+    std::cout<< "ov_cpu_verbose,exec,cpu,"
```

Comment:
cpu -> cpu / dnnl / ref

Reply:
ov_cpu_verbose -> can we align with gpu?

Reply:
@dmitry-gorokhov As for **cpu** or **dnnl**, I don't see a good was to distinguish this at the moment.
I really don't want to introduce a separate method in order to access *prim* field, and this could be not complete solution as we discussed.
One of the possible implementation is to group up node types, so, for example, dnnl types come first and cpu types come after (or vice versa).
Also as far as I remember we will soon distinguish dnnl nodes from cpu nodes in scope of dynamic shapes support.

Reply:
Can we mark Node and Verbose classes as friends? It allow to access prim field. We can consider such solution until MKLDNNNode refactoring will not be completed.

Reply:
I was considering this option.
Didn't want to add DEBUG_CAPS "#ifdef" to MKLDNNNode
But of course we could do it.
So we chose this option then?

Reply:
Done